Source/Core/Editor/UI/Windows/SearchableTagListWindow.cs

```diff
@@ -0,0 +1,311 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.UIElements;
+using UnityEngine;
+using UnityEngine.UIElements;
+using VRBuilder.Core.SceneObjects;
+using VRBuilder.Core.Settings;
+
+namespace VRBuilder.Editor.UI.Windows
+{
+    public class SearchableTagListWindow : EditorWindow
+    {
+        /// <summary>
+        /// Root VisualTreeAsset for the searchable list.
+        /// </summary>
+        [SerializeField]
+        private VisualTreeAsset selectItemsPanel = default;
+
+        /// <summary>
+        /// VisualTreeAsset for the individual list items.
+        /// </summary>
+        [SerializeField]
+        private VisualTreeAsset listItem = default;
+
+        /// <summary>
+        /// The size of the popup window.
+        /// </summary>
+        private Vector2 windowSize;
+
+        /// <summary>
+        /// The minimum size of the popup window.
+        /// </summary>
+        private Vector2 minWindowSize = new Vector2(200, 200);
+
+        /// <summary>
+        /// The ScrollView containing the list of tags.
+        /// </summary>
+        private ScrollView tagScrollView;
+
+        /// <summary>
+        /// The list of tags which will be used.
+        /// </summary>
+        private List<SceneObjectTags.Tag> availableTags;
+
+        /// <summary>
+        /// The list of tags which will be used.
+        /// </summary>
+        private List<SceneObjectTags.Tag> selectedTags;
+
+        /// <summary>
+        /// Callback to invoke when a tag is selected.
+        /// </summary>
+        private Action<List<SceneObjectTags.Tag>> onItemSelected;
+
+        public void SetItemsSelectedCallBack(Action<List<SceneObjectTags.Tag>> onItemSelected)
+        {
+            this.onItemSelected = onItemSelected;
+            windowSize = minWindowSize;
+        }
+
+        public Vector2 GetWindowSize()
+        {
+            return windowSize;
+        }
+
+        public void OnGUI()
+        {
+            // intentionally left blank
```

Comment:
Does ongui have to be implemented at all?

Reply:
Will remove it